pkg/nexus/common/scheduler/base-nexus-scheduler.go

```diff
@@ -24,19 +24,19 @@ type BaseNexusScheduler struct {
 	client     *http.Client
 }
 
-func NewBaseNexusScheduler(queue *queue.NexusQueue, config *config.BaseNexusSchedulerConfig, nexusConfig *config.NexusConfig) *BaseNexusScheduler {
+func NewBaseNexusScheduler(queue *queue.NexusQueue, config *config.BaseNexusSchedulerConfig, nexusConfig *config.NexusConfig, client *http.Client) *BaseNexusScheduler {
 	return &BaseNexusScheduler{
 		BaseNexusSchedulerConfig: config,
 		Queue:                    queue,
 		requestUrl:               models.NUCLIO_NEXUS_REQUEST_URL,
-		client:                   &http.Client{},
+		client:                   client,
```

Comment:
WIeso muss dieser übergeben werden?

Reply:
damit ich in den tests den custom http.Client mit dem RoundTrip übergeben kann, dadurch kann man das besser testen bzw. hat man dann nicht mehr das problem, das die anfragen errors werfen,